src/manifold/src/manifold.cpp

```diff
@@ -71,6 +73,67 @@ Manifold Halfspace(Box bBox, glm::vec3 normal, float originOffset) {
   float zDeg = glm::degrees(glm::atan(normal.y, normal.x));
   return cutter.Rotate(0.0f, yDeg, zDeg);
 }
+
+double DistanceToVector(const glm::vec3& pt, const glm::vec3& v) {
+  return glm::length(pt - v * glm::dot(pt, v));
+}
+
+inline double DistanceToPlane(const glm::vec3& pt, const glm::vec4& plane) {
+  return glm::dot({plane.x, plane.y, plane.z}, pt) - plane.w;
+}
+
+glm::vec4 PlaneOfPts(const glm::vec3& a, const glm::vec3& b,
+                     const glm::vec3& c) {
+  auto crx = glm::cross(c - a, b - a);
+  auto n = glm::length(crx);
+  if (n == 0.) return glm::vec4();
+  return glm::vec4(crx, glm::dot(crx, a)) / n;
+}
+
+// Return a trio of non-collinear indices into pts. If none are found, indices
+// will be -1 (and hulling cannot be performed).
+glm::ivec3 NonCollinearTriple(const std::vector<glm::vec3>& pts,
+                              const double precision) {
+  const int len = pts.size();
+  int furthest = 1;
+  float dist = glm::distance(pts[0], pts[1]);
```

Comment:
Also, if all you care about is the max/min, always used squared distance (dot-product) since it's much faster than a sqrt.